google/cloud/bigquery/client.py

```diff
@@ -1367,13 +1364,13 @@ def create_job(self, job_config, retry=DEFAULT_RETRY):
             extract_job_config = google.cloud.bigquery.job.ExtractJobConfig.from_api_repr(
                 job_config
             )
-            source = TableReference.from_api_repr(job_config["extract"]["sourceTable"])
+            source = _get_sub_prop(job_config, ["extract", "sourceTable"])
             destination_uris = _get_sub_prop(job_config, ["extract", "destinationUris"])
             return self.extract_table(
                 source, destination_uris, job_config=extract_job_config, retry=retry
             )
         elif "query" in job_config:
-            del job_config["query"]["destinationTable"]
+            _del_sub_prop(job_config, ["query", "destinationTable"])
```

Comment:
This will still potentially modify the input argument in-place, do we mind? (cc: @tswast )

Reply:
Let's not do that if at all possible. Maybe make a copy before calling `_del_sub_prop`?

Reply:
@plamut , @tswast  Ok, may i do `copy_config = copy.deepcopy(job_config)` before calling `_del_sub_prop` ?

If  i consider `clear destination_table if it was a query job`  statement which mentioned in issue's description, need to modify the input argument.

Reply:
@HemangChothani Creating a copy and operating on the latter sounds good, yes.

> If i consider `clear destination_table if it was a query job` statement which mentioned in issue's description, need to modify the input argument.

If I understood the ticket description correctly, it is not clear when a client library should clear the destination table property. But if we do clear it (as is the case with `query` jobs), it is safer to do it in a config copy, because users normally don't expect that their input parameter could be modified.